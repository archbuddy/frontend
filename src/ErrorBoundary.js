import React from 'react'
import { log } from './util'

// https://reactjs.org/docs/error-boundaries.html
export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    log(error)
    return { hasError: true }
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
    log(error, errorInfo)
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div id="errorPage">
          <div className="errorPageHeader">Arch Buddy</div>
          <div className="errorPageContent">
            <p>Ooppsss... Something has happened</p>
            <p>Try to refresh the page</p>
          </div>
        </div>
      )
    }
    if (this.props.children) {
      return this.props.children
    }
    return (
      <div id="errorPage">
        <div className="errorPageHeader">Arch Buddy</div>
        <div className="errorPageContent">
          <p>Ooppsss... Something has happened</p>
          <p>This is a leak :(</p>
        </div>
      </div>
    )
  }
}
