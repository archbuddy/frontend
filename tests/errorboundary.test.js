import renderer from 'react-test-renderer'
import ErrorBoundary from '../src/ErrorBoundary'

test('render ErrorBoundary', () => {
  const component = renderer.create(<ErrorBoundary />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('render ErrorBoundary - with children', () => {
  const component = renderer.create(
    <ErrorBoundary>
      <h1>text</h1>
    </ErrorBoundary>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
