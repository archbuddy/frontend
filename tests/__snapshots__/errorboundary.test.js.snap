// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ErrorBoundary - with children 1`] = `
<h1>
  text
</h1>
`;

exports[`render ErrorBoundary 1`] = `
<div
  id="errorPage"
>
  <div
    className="errorPageHeader"
  >
    Arch Buddy
  </div>
  <div
    className="errorPageContent"
  >
    <p>
      Ooppsss... Something has happened
    </p>
    <p>
      This is a leak :(
    </p>
  </div>
</div>
`;
